Running WASM without CrossOriginIsolation

### Describe the bug

This is a followup issue to the discussion https://github.com/fzyzcjy/flutter_rust_bridge/discussions/2401 that was answered previously.

And related to the threadpool issue in #2215 

From the prior discussions and the issue related it seems possible to compile FRB without any async functions and to disable the two cross origin headers https://cjycode.com/flutter_rust_bridge/manual/miscellaneous/web-cross-origin

In theory, from what I gather this should be as simple as making sure all my functions in my code are `pub fn` and I either mark all of them with `#[frb(sync)]` or use `default_dart_async: false` in my flutter_rust_bridge.yml config file. Then if I remove the two headers for `require-corp` and `same-origin` it should be ok, as it is all running under the same thread not trying to share buffers.

But doing this and compiling using `flutter_rust_bridge_codegen generate` and `dart run flutter_rust_bridge build-web --dart-root $TMS_FLUTTER --rust-root $TMS_INFRA` doesn't yield any results. And it's providing me with an exception and a blank white screen if I try host the flutter app.

```
Uncaught : Buffers cannot be shared due to missing cross-origin headers. Please refer to https://fzyzcjy.github.io/flutter_rust_bridge/manual/miscellaneous/web-cross-origin for details.
```

Doing a bit of digging I've found that when the FRB code initializes the Wasm modules it simply throws this exception if it doesn't detect the headers.
```dart
/// {@macro flutter_rust_bridge.internal}
Future<void> initializeWasmModule({required String root}) async {
  _ensureCrossOriginIsolated();

  final script = web.HTMLScriptElement()..src = '$root.js';
  web.document.head!.append(script);

  await script.onLoad.first;

  jsEval('window.wasm_bindgen = wasm_bindgen');

  await _jsWasmBindgen('${root}_bg.wasm').toDart;
}

@JS('wasm_bindgen')
external JSPromise _jsWasmBindgen(String path);

void _ensureCrossOriginIsolated() {
  switch (crossOriginIsolated) {
    case false:
      throw const MissingHeaderException();
    case true:
      return;
    case null:
      web.console.warn(
          'Warning: crossOriginIsolated is null, browser might not support buffer sharing.'
              .toJS);
      return;
  }
}
```
I tried disabling the exception quickly just to test if it would continue and work as I expect it should, or if there actually was a problem. And it definitely has some kind of issue with my code
```
panicked at tms_infra/src/frb_generated.rs:8370:52:
called `Result::unwrap()` on an `Err` value: Error { kind: UnexpectedEof, message: "failed to fill whole buffer" }

Stack:

__wbg_get_imports/imports.wbg.__wbg_new_abda76e883ba8a5f/<@http://localhost:8080/ui/pkg/tms_infra.js:558:21
logError@http://localhost:8080/ui/pkg/tms_infra.js:243:18
__wbg_get_imports/imports.wbg.__wbg_new_abda76e883ba8a5f@http://localhost:8080/ui/pkg/tms_infra.js:557:66
tms_infra.wasm.console_error_panic_hook::Error::new::he028879d8df251be@http://localhost:8080/ui/pkg/tms_infra_bg.wasm:wasm-function[21333]:0x88caed
tms_infra.wasm.console_error_panic_hook::hook_impl::h9b796c2c63ed13bc@http://localhost:8080/ui/pkg/tms_infra_bg.wasm:wasm-function[5417]:0x5c5366
tms_infra.wasm.console_error_panic_hook::hook::h45812836ec7bc703@http://localhost:8080/ui/pkg/tms_infra_bg.wasm:wasm-function[29404]:0x90dd32
tms_infra.wasm.core::ops::function::Fn::call::hdaf9a228a854a838@http://localhost:8080/ui/pkg/tms_infra_bg.wasm:wasm-…
```
I'm not certain what I'm actually doing wrong here, or what I specifically need to do to stop FRB from sharing buffers and requiring either async, or needing https.

On a side note, it would be quite interesting to have some kind of config value similar to `default_dart_async: false` but for the cross origin, to allow only synchronous that doesn't need to run with the headers. `allow_unsafe_http: true` or `no_shared_buffers: true` etc... Rather than some kind of hack to get http to working for FRB WASM. But it is a niche use case.

### Steps to reproduce
1. clone and modify the https://github.com/fzyzcjy/flutter_rust_bridge/tree/master/frb_example/flutter_via_create
2. Modify and change `frb_dart/lib/src/wasm_module/_web.dart` with the following
```dart
void _ensureCrossOriginIsolated() {
  switch (crossOriginIsolated) {
    case false:
      // throw const MissingHeaderException();
      return; // just return here to stop the exeception popping up
    case true:
      return;
    case null:
      web.console.warn('Warning: crossOriginIsolated is null, browser might not support buffer sharing.'.toJS);
      return;
  }
}
```
3. Set `default_dart_async: false` in flutter_rust_bridge.yaml`
4. Remove all async and keep all functions as `pub fn`
5. Compile rust code for wasm32-unknown-unknown (sometimes needed in my case)
6. Compile project for wasm/web using `dart run flutter_rust_bridge build-web`
7. Run flutter using `flutter run -d chrome`


### Logs

```shell
Uncaught : Buffers cannot be shared due to missing cross-origin headers. Please refer to https://fzyzcjy.github.io/flutter_rust_bridge/manual/miscellaneous/web-cross-origin for details.
****

panicked at tms_infra/src/frb_generated.rs:8370:52:
called `Result::unwrap()` on an `Err` value: Error { kind: UnexpectedEof, message: "failed to fill whole buffer" }

Stack:

__wbg_get_imports/imports.wbg.__wbg_new_abda76e883ba8a5f/<@http://localhost:8080/ui/pkg/tms_infra.js:558:21
logError@http://localhost:8080/ui/pkg/tms_infra.js:243:18
__wbg_get_imports/imports.wbg.__wbg_new_abda76e883ba8a5f@http://localhost:8080/ui/pkg/tms_infra.js:557:66
tms_infra.wasm.console_error_panic_hook::Error::new::he028879d8df251be@http://localhost:8080/ui/pkg/tms_infra_bg.wasm:wasm-function[21333]:0x88caed
tms_infra.wasm.console_error_panic_hook::hook_impl::h9b796c2c63ed13bc@http://localhost:8080/ui/pkg/tms_infra_bg.wasm:wasm-function[5417]:0x5c5366
tms_infra.wasm.console_error_panic_hook::hook::h45812836ec7bc703@http://localhost:8080/ui/pkg/tms_infra_bg.wasm:wasm-function[29404]:0x90dd32
tms_infra.wasm.core::ops::function::Fn::call::hdaf9a228a854a838@http://localhost:8080/ui/pkg/tms_infra_bg.wasm:wasm-function[24851]:0x8ce1f1
tms_infra.wasm.<alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call::h488a605a49d40355@http://localhost:8080/ui/pkg/tms_infra_bg.wasm:wasm-function[21078]:0x887251
tms_infra.wasm.std::panicking::rust_panic_with_hook::hf8f6578efc75fc6a@http://localhost:8080/ui/pkg/tms_infra_bg.wasm:wasm-function[280]:0x1c9a97
tms_infra.wasm.std::panicking::begin_panic_handler::{{closure}}::h3e12cffce08c1729@http://localhost:8080/ui/pkg/tms_infra_bg.wasm:wasm-function[5197]:0x5af003
tms_infra.wasm.std::sys::backtrace::__rust_end_short_backtrace::haebcb3602c9b22fb@http://localhost:8080/ui/pkg/tms_infra_bg.wasm:wasm-function[31700]:0x926a58
tms_infra.wasm.rust_begin_unwind@http://localhost:8080/ui/pkg/tms_infra_bg.wasm:wasm-function[15746]:0x7fcade
tms_infra.wasm.core::panicking::panic_fmt::haabf28ee12d2afe3@http://localhost:8080/ui/pkg/tms_infra_bg.wasm:wasm-function[17396]:0x82dce5
tms_infra.wasm.core::result::unwrap_failed::haf5ec27a87751d48@http://localhost:8080/ui/pkg/tms_infra_bg.wasm:wasm-function[5546]:0x5d17b8
tms_infra.wasm.<u32 as tms_infra::frb_generated::SseDecode>::sse_decode::h79fbdac1200ba067@http://localhost:8080/ui/pkg/tms_infra_bg.wasm:wasm-function[10822]:0x7335cf
tms_infra.wasm.<tms_infra::infra::database_schemas::tms_time::tms_date::TmsDate as tms_infra::frb_generated::SseDecode>::sse_decode::habc389615d0ba933@http://localhost:8080/ui/pkg/tms_infra_bg.wasm:wasm-function[18206]:0x843e26
tms_infra.wasm.tms_infra::frb_generated::wire__crate__infra__database_schemas__tms_time__tms_date__tms_date_is_same_moment_impl::{{closure}}::h3d7fb6bf71a76d23@http://localhost:8080/ui/pkg/tms_infra_bg.wasm:wasm-function[3005]:0x4a3867
tms_infra.wasm.<flutter_rust_bridge::handler::implementation::executor::SimpleExecutor<EL,TP,AR> as flutter_rust_bridge::handler::executor::Executor>::execute_sync::h17d4cc3010be4cef@http://localhost:8080/ui/pkg/tms_infra_bg.wasm:wasm-function[4878]:0x58dbd3
tms_infra.wasm.<flutter_rust_bridge::handler::implementation::handler::SimpleHandler<E,EL> as flutter_rust_bridge::handler::handler::Handler>::wrap_sync::{{closure}}::{{closure}}::hd2e65ad388a17866@http://localhost:8080/ui/pkg/tms_infra_bg.wasm:wasm-function[7227]:0x658652
tms_infra.wasm.<core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once::h690461cc09197fc0@http://localhost:8080/ui/pkg/tms_infra_bg.wasm:wasm-function[8080]:0x68f930
tms_infra.wasm.std::panicking::try::do_call::hd4a9fa85ce53609d@http://localhost:8080/ui/pkg/tms_infra_bg.wasm:wasm-function[8734]:0x6b87b1
tms_infra.wasm.std::panicking::try::hd16b149fe8c539e2@http://localhost:8080/ui/pkg/tms_infra_bg.wasm:wasm-function[1967]:0x401f4d
tms_infra.wasm.std::panic::catch_unwind::h9b6d657d96a441f3@http://localhost:8080/ui/pkg/tms_infra_bg.wasm:wasm-function[32951]:0x92d702
tms_infra.wasm.flutter_rust_bridge::misc::panic_backtrace::PanicBacktrace::catch_unwind::hd8a452ad937f5278@http://localhost:8080/ui/pkg/tms_infra_bg.wasm:wasm-function[20392]:0x877e38
tms_infra.wasm.<flutter_rust_bridge::handler::implementation::handler::SimpleHandler<E,EL> as flutter_rust_bridge::handler::handler::Handler>::wrap_sync::{{closure}}::hc0e56ca4e82afe5a@http://localhost:8080/ui/pkg/tms_infra_bg.wasm:wasm-function[2843]:0x48baaa
tms_infra.wasm.<core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once::h59716dfdd44d06db@http://localhost:8080/ui/pkg/tms_infra_bg.wasm:wasm-function[7696]:0x677a35
tms_infra.wasm.std::panicking::try::do_call::hb0a14a5d088ff73a@http://localhost:8080/ui/pkg/tms_infra_bg.wasm:wasm-function[16768]:0x81b4f3
tms_infra.wasm.std::panicking::try::hf32cc74fce528fcb@http://localhost:8080/ui/pkg/tms_infra_bg.wasm:wasm-function[3734]:0x50406c
tms_infra.wasm.std::panic::catch_unwind::hd1f6d561170c04dc@http://localhost:8080/ui/pkg/tms_infra_bg.wasm:wasm-function[33097]:0x92dd48
tms_infra.wasm.<flutter_rust_bridge::handler::implementation::handler::SimpleHandler<E,EL> as flutter_rust_bridge::handler::handler::Handler>::wrap_sync::h65bd50e370dd9a34@http://localhost:8080/ui/pkg/tms_infra_bg.wasm:wasm-function[3223]:0x4c0ca6
tms_infra.wasm.tms_infra::frb_generated::wire__crate__infra__database_schemas__tms_time__tms_date__tms_date_is_same_moment_impl::h3bd9fb6cf97ff55b@http://localhost:8080/ui/pkg/tms_infra_bg.wasm:wasm-function[10006]:0x7056cb
tms_infra.wasm.tms_infra::frb_generated::pde_ffi_dispatcher_sync_impl::h629dc62aa1a2bb0e@http://localhost:8080/ui/pkg/tms_infra_bg.wasm:wasm-function[101]:0xec92a
tms_infra.wasm.tms_infra::frb_generated::web::frb_pde_ffi_dispatcher_sync::h7cf03248177e23f7@http://localhost:8080/ui/pkg/tms_infra_bg.wasm:wasm-function[21766]:0x895a35
tms_infra.wasm.frb_pde_ffi_dispatcher_sync@http://localhost:8080/ui/pkg/tms_infra_bg.wasm:wasm-function[10879]:0x7365a1
__exports.frb_pde_ffi_dispatcher_sync@http://localhost:8080/ui/pkg/tms_infra.js:295:22
bj@http://localhost:8080/ui/main.dart.js:33109:23
$0@http://localhost:8080/ui/main.dart.js:103048:5
aCw@http://localhost:8080/ui/main.dart.js:96255:11
b2@http://localhost:8080/ui/main.dart.js:96262:13
aAG@http://localhost:8080/ui/main.dart.js:102525:12
Nu/$async$Nu<@http://localhost:8080/ui/main.dart.js:102498:39
o/s@http://localhost:8080/ui/main.dart.js:4647:64
$2@http://localhost:8080/ui/main.dart.js:42091:14
m@http://localhost:8080/ui/main.dart.js:4633:10
Nu@http://localhost:8080/ui/main.dart.js:102500:10
aEI/$async$yJ<@http://localhost:8080/ui/main.dart.js:96239:14
o/s@http://localhost:8080/ui/main.dart.js:4647:64
$2@http://localhost:8080/ui/main.dart.js:42091:14
$1@http://localhost:8080/ui/main.dart.js:42085:21
aJl@http://localhost:8080/ui/main.dart.js:43117:34
Pu@http://localhost:8080/ui/main.dart.js:43120:13
$0@http://localhost:8080/ui/main.dart.js:42478:11
Bm@http://localhost:8080/ui/main.dart.js:4759:40
rY@http://localhost:8080/ui/main.dart.js:42413:3
cI@http://localhost:8080/ui/main.dart.js:42080:8
l@http://localhost:8080/ui/main.dart.js:4637:10
BS/$async$BS<@http://localhost:8080/ui/main.dart.js:96250:17
o/s@http://localhost:8080/ui/main.dart.js:4647:64
$2@http://localhost:8080/ui/main.dart.js:42091:14
$1@http://localhost:8080/ui/main.dart.js:42085:21
aJl@http://localhost:8080/ui/main.dart.js:43117:34
Pu@http://localhost:8080/ui/main.dart.js:43120:13
$0@http://localhost:8080/ui/main.dart.js:42478:11
Bm@http://localhost:8080/ui/main.dart.js:4759:40
rY@http://localhost:8080/ui/main.dart.js:42413:3
cI@http://localhost:8080/ui/main.dart.js:42080:8
l@http://localhost:8080/ui/main.dart.js:4637:10
aZT/$async$aZT<@http://localhost:8080/ui/main.dart.js:33124:17
o/s@http://localhost:8080/ui/main.dart.js:4647:64
$2@http://localhost:8080/ui/main.dart.js:42091:14
$1@http://localhost:8080/ui/main.dart.js:42085:21
aJl@http://localhost:8080/ui/main.dart.js:43117:34
Pu@http://localhost:8080/ui/main.dart.js:43120:13
$0@http://localhost:8080/ui/main.dart.js:42478:11
Bm@http://localhost:8080/ui/main.dart.js:4759:40
rY@http://localhost:8080/ui/main.dart.js:42413:3
cI@http://localhost:8080/ui/main.dart.js:42080:8
l@http://localhost:8080/ui/main.dart.js:4637:10
a9I/$async$a9I<@http://localhost:8080/ui/main.dart.js:33138:17
o/s@http://localhost:8080/ui/main.dart.js:4647:64
$2@http://localhost:8080/ui/main.dart.js:42091:14
$1@http://localhost:8080/ui/main.dart.js:42085:21
aJl@http://localhost:8080/ui/main.dart.js:43117:34
Pu@http://localhost:8080/ui/main.dart.js:43120:13
$0@http://localhost:8080/ui/main.dart.js:42478:11
Bm@http://localhost:8080/ui/main.dart.js:4759:40
rY@http://localhost:8080/ui/main.dart.js:42413:3
$0@http://localhost:8080/ui/main.dart.js:42449:13
br5@http://localhost:8080/ui/main.dart.js:4791:5
brm@http://localhost:8080/ui/main.dart.js:4793:7
$1@http://localhost:8080/ui/main.dart.js:42036:3
MutationCallback*bnw@http://localhost:8080/ui/main.dart.js:4616:1
@http://localhost:8080/ui/main.dart.js:120210:23
lazyFinal/a[c]@http://localhost:8080/ui/main.dart.js:24:36
bb4@http://localhost:8080/ui/main.dart.js:4798:13
pj@http://localhost:8080/ui/main.dart.js:4890:3
SR@http://localhost:8080/ui/main.dart.js:42420:3
kT@http://localhost:8080/ui/main.dart.js:42418:13
cI@http://localhost:8080/ui/main.dart.js:42077:13
l@http://localhost:8080/ui/main.dart.js:4637:10
$0/$async$$0<@http://localhost:8080/ui/main.dart.js:35027:10
o/s@http://localhost:8080/ui/main.dart.js:4647:64
$2@http://localhost:8080/ui/main.dart.js:42091:14
m@http://localhost:8080/ui/main.dart.js:4633:10
$0@http://localhost:8080/ui/main.dart.js:35028:10
nr/$async$nr<@http://localhost:8080/ui/main.dart.js:34834:28
o/s@http://localhost:8080/ui/main.dart.js:4647:64
$2@http://localhost:8080/ui/main.dart.js:42091:14
m@http://localhost:8080/ui/main.dart.js:4633:10
nr@http://localhost:8080/ui/main.dart.js:34838:10
$0/$async$$0<@http://localhost:8080/ui/main.dart.js:36186:38
o/s@http://localhost:8080/ui/main.dart.js:4647:64
$2@http://localhost:8080/ui/main.dart.js:42091:14
m@http://localhost:8080/ui/main.dart.js:4633:10
$0@http://localhost:8080/ui/main.dart.js:36190:10
bt6/$async$aZK<@http://localhost:8080/ui/main.dart.js:912:34
o/s@http://localhost:8080/ui/main.dart.js:4647:64
$2@http://localhost:8080/ui/main.dart.js:42091:14
m@http://localhost:8080/ui/main.dart.js:4633:10
bt6@http://localhost:8080/ui/main.dart.js:915:10
aZK@http://localhost:8080/ui/main.dart.js:889:17
a5T/$async$$1<@http://localhost:8080/ui/main.dart.js:46275:14
o/s@http://localhost:8080/ui/main.dart.js:4647:64
$2@http://localhost:8080/ui/main.dart.js:42091:14
m@http://localhost:8080/ui/main.dart.js:4633:10
a5T@http://localhost:8080/ui/main.dart.js:46277:10
$1@http://localhost:8080/ui/main.dart.js:46267:19
a5D/$async$$1<@http://localhost:8080/ui/main.dart.js:33535:16
o/s@http://localhost:8080/ui/main.dart.js:4647:64


tms_infra.js:552:21
```


### Expected behavior

Should put all the code in the WASM module onto a single thread, and when hosted using http. It shouldn't need the two headers which force https on the user.

### Generated binding code

_No response_

### OS

PopOS & Firefox Browser

### Version of `flutter_rust_bridge_codegen`

2.6.0

### Flutter info

```shell
[✓] Flutter (Channel stable, 3.24.4, on Pop!_OS 22.04 LTS 6.9.3-76060903-generic, locale en_AU.UTF-8)
    • Flutter version 3.24.4 on channel stable at /home/cj/snap/flutter/common/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 603104015d (4 weeks ago), 2024-10-24 08:01:25 -0700
    • Engine revision db49896cf2
    • Dart version 3.5.4
    • DevTools version 2.37.3

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /home/cj/Android/Sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /home/cj/Android/Sdk
    • Java binary at: /home/cj/Downloads/android-studio/jbr/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.9+0-17.0.9b1087.7-11185874)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • clang version 10.0.0-4ubuntu1
    • cmake version 3.16.3
    • ninja version 1.10.0
    • pkg-config version 0.29.1

[✓] Android Studio (version 2023.2)
    • Android Studio at /home/cj/Downloads/android-studio
    • Flutter plugin version 78.4.1
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.9+0-17.0.9b1087.7-11185874)

[✓] VS Code (version 1.95.3)
    • VS Code at /usr/share/code
    • Flutter extension version 3.100.0

[✓] Connected device (2 available)
    • Linux (desktop) • linux  • linux-x64      • Pop!_OS 22.04 LTS 6.9.3-76060903-generic
    • Chrome (web)    • chrome • web-javascript • Google Chrome 131.0.6778.69

[✓] Network resources
    • All expected network resources are available.
```


### Version of `clang++`

_No response_

### Additional context

_No response_

Hmmmm, so this is currently working, not just in the frb_via_create but it's also started to work on my actual application as well. Which is crazy. The only thing I can think of which is different is likely my device having issues with compiling the WASM. Which is related to older issue I opened a while ago #2161 

Where on occasion on certain linux based devices it just doesn't like compiling with the command `dart run flutter_rust_bridge build-web` or `flutter_rust_bridge_codegen build-web`. Strangely I've never noticed it pop up in my CI-CD. Just on my laptop and my Desktop. And to this day I've never really figured out why. Usually resulting in an error like below
```
Building wasm for /home/cj/Documents/code/TMS/tms_infra
> which wasm-pack (pwd: null, env: null)
/home/cj/.cargo/bin/wasm-pack
> cargo read-manifest (pwd: /home/cj/Documents/code/TMS/tms_infra, env: null)
> wasm-pack build -t no-modules -d /home/cj/Documents/code/TMS/tms_client/web/pkg --no-typescript --out-name tms_infra --dev /home/cj/Documents/code/TMS/tms_infra -- -Z build-std=std,panic_abort (pwd: null, env: {RUSTUP_TOOLCHAIN: nightly, RUSTFLAGS: -C target-feature=+atomics,+bulk-memory,+mutable-globals, CARGO_TERM_COLOR: always})
[INFO]: 🎯  Checking for the Wasm target...
[INFO]: 🌀  Compiling to Wasm...
   Compiling compiler_builtins v0.1.134
   Compiling core v0.0.0 (/home/cj/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core)
   Compiling libc v0.2.161
   Compiling memchr v2.5.0
   Compiling dlmalloc v0.2.6
   Compiling std v0.0.0 (/home/cj/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std)
   Compiling proc-macro2 v1.0.81
   Compiling unicode-ident v1.0.12
   Compiling wasm-bindgen-shared v0.2.95
   Compiling autocfg v1.2.0
   Compiling bumpalo v3.16.0
   Compiling log v0.4.22
   Compiling typenum v1.17.0
   Compiling version_check v0.9.4
   Compiling wasm-bindgen v0.2.95
   Compiling syn v1.0.109
   Compiling serde v1.0.198
   Compiling anyhow v1.0.82
   Compiling portable-atomic v1.9.0
   Compiling rustversion v1.0.18
   Compiling serde_json v1.0.116
error: linking with `cc` failed: exit status: 1
  |
  = note: LC_ALL="C" PATH="/home/cj/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin:/home/cj/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/self-contained:/snap/flutter/current/usr/bin:/snap/flutter/current/bin:/home/cj/snap/flutter/common/flutter/bin:/snap/flutter/current/usr/bin:/snap/flutter/current/bin:/home/cj/snap/flutter/common/flutter/bin:/home/cj/.local/share/pnpm:/home/cj/.nvm/versions/node/v21.7.1/bin:/home/cj/.cargo/bin:/home/cj/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin:/home/cj/.local/bin:/home/cj/.local/bin" VSLANG="1033" "cc" "-m64" "/tmp/rustcKs45UW/symbols.o" "/home/cj/Documents/code/TMS/target/debug/build/wasm-bindgen-91217f100852db63/build_script_build-91217f100852db63.build_script_build.af0d2c80bf26ac47-cgu.0.rcgu.o" "/home/cj/Documents/code/TMS/target/debug/build/wasm-bindgen-91217f100852db63/build_script_build-91217f100852db63.07g1gcqp3e0ikddp1yld65hv3.rcgu.o" "-Wl,--as-needed" "-Wl,-Bstatic" "/home/cj/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-fb8c387236db3c10.rlib" "/home/cj/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libpanic_unwind-b251837809b2f64b.rlib" "/home/cj/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libobject-cbef83eaa4e01f04.rlib" "/home/cj/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libmemchr-12a772286de2fca7.rlib" "/home/cj/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libaddr2line-023bbe5ada7abfc3.rlib" "/home/cj/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libgimli-de2bcdf10ec69c2b.rlib" "/home/cj/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_demangle-8e1754df350800fd.rlib" "/home/cj/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd_detect-06c2ab7b9c32e037.rlib" "/home/cj/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libhashbrown-51d2621fc57a281b.rlib" "/home/cj/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_std_workspace_alloc-54a930408793772f.rlib" "/home/cj/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libminiz_oxide-0c7d01cba400dcb2.rlib" "/home/cj/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libadler-b9f84d69e331b75b.rlib" "/home/cj/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libunwind-eec95493246aae39.rlib" "/home/cj/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcfg_if-6f6676c475e08ab0.rlib" "/home/cj/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liblibc-799d520624f8b2e0.rlib" "/home/cj/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liballoc-fcf93d4842acd9a9.rlib" "/home/cj/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_std_workspace_core-395a38b8e0851c9b.rlib" "/home/cj/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcore-d453bab70303062c.rlib" "/home/cj/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-0e6386368b0a24e8.rlib" "-Wl,-Bdynamic" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "-B/home/cj/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/gcc-ld" "-fuse-ld=lld" "-Wl,--eh-frame-hdr" "-Wl,-z,noexecstack" "-L" "/home/cj/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-o" "/home/cj/Documents/code/TMS/target/debug/build/wasm-bindgen-91217f100852db63/build_script_build-91217f100852db63" "-Wl,--gc-sections" "-pie" "-Wl,-z,relro,-z,now" "-nodefaultlibs"
  = note: rust-lld: error: undefined reference: _dl_find_object@GLIBC_2.35
          >>> referenced by /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libgcc_s.so.1 (disallowed by --no-allow-shlib-undefined)
          collect2: error: ld returned 1 exit status
```

A hack that i've found is to run a raw wasm-pack build. And then the frb codegen build-web command. Which for some magical reason works. So I keep a build script on hand to just do that each time I need to build to wasm
```sh
#
# TMS INFRA bindings
#
TMS_INFRA=$(pwd)/tms_infra
TMS_FLUTTER=$(pwd)/tms_client

# codegen for flutter (green text)
echo -e "${G}Running codegen for $TMS_INFRA"
(cd $TMS_FLUTTER && flutter_rust_bridge_codegen generate)
# build wasm logic
echo -e "${G}Building wasm for $TMS_INFRA"
wasm-pack build $TMS_INFRA --target no-modules --out-dir $TMS_FLUTTER/web/pkg --no-typescript --out-name tms_infra --dev -- -Z build-std=std,panic_abort
(cd $TMS_FLUTTER && dart run flutter_rust_bridge build-web --dart-root $TMS_FLUTTER --rust-root $TMS_INFRA)
```

So before I run any flutter_rust_bridge commands I always have to "pre-compile" my code with wasm-pack first just to kick it in the right direction (which I left out of the reproducible steps because it's a very isolated issue on linux on only a small number of devices). 

But I'm stunned that this is working now. And I have no idea why. I mean this is great. But I've got no idea what's happened since yesterday and today, where both the reproducible steps and my application could not run. And now they both work fine.

I'll keep monitoring it for the time being, but hopefully this was just a fluke.
The only thing I think is needed now for this to work in the long term, is someway to bypass the exception thrower caused by checking the response headers. Other than manually changing the library code. Some kind of option maybe?

--- a/frb_dart/lib/src/exceptions.dart
+++ b/frb_dart/lib/src/exceptions.dart
@@ -31,20 +31,6 @@ abstract class FrbBacktracedException extends FrbException {
   String get backtrace;
 }
 
-/// Thrown when the browser is not run in a [cross-origin isolated] environment.
-///
-/// [cross-origin isolated]: https://developer.mozilla.org/en-US/docs/Web/API/crossOriginIsolated
-class MissingHeaderException implements FrbException {
-  /// Constructs an exception
-  const MissingHeaderException();
-
-  static const _message =
-      '''Buffers cannot be shared due to missing cross-origin headers. Please refer to https://fzyzcjy.github.io/flutter_rust_bridge/manual/miscellaneous/web-cross-origin for details.''';
-
-  @override
-  String toString() => _message;
-}
-
 /// Platform is not matched
 class PlatformMismatchException implements FrbException {
   /// Constructs an exception

--- a/frb_dart/lib/src/wasm_module/_web.dart
+++ b/frb_dart/lib/src/wasm_module/_web.dart
@@ -1,7 +1,6 @@
 import 'dart:async';
 import 'dart:js_interop';
 
-import 'package:flutter_rust_bridge/src/exceptions.dart';
 import 'package:flutter_rust_bridge/src/platform_utils/_web.dart';
 import 'package:web/web.dart' as web;
 
@@ -25,7 +24,10 @@ external JSPromise _jsWasmBindgen(String path);
 void _ensureCrossOriginIsolated() {
   switch (crossOriginIsolated) {
     case false:
-      throw const MissingHeaderException();
+      web.console.warn(
+          'Warning: Buffers cannot be shared due to missing cross-origin headers. Please refer to https://fzyzcjy.github.io/flutter_rust_bridge/manual/miscellaneous/web-cross-origin for details.'
+              .toJS);
+      return;
     case true:
       return;
     case null:

--- a/frb_dart/test/exceptions_test.dart
+++ b/frb_dart/test/exceptions_test.dart
@@ -10,11 +10,6 @@ void main() {
     expect(AnyhowException('hello').toString(), contains('AnyhowException'));
   });
 
-  test('MissingHeaderException', () {
-    expect(const MissingHeaderException().toString(),
-        contains('cannot be shared'));
-  });
-
   test('PlatformMismatchException', () {
     expect(const PlatformMismatchException().toString(),
         contains('Not implemented on'));

--- a/website/docs/manual/miscellaneous/02-web-cross-origin.md
+++ b/website/docs/manual/miscellaneous/02-web-cross-origin.md
@@ -60,7 +60,7 @@ Thanks to [this pull request](https://github.com/flutter/flutter/pull/136297), w
 flutter drive --web-header=Cross-Origin-Opener-Policy=same-origin --web-header=Cross-Origin-Embedder-Policy=require-corp
 ```
 
-## When `flutter run` without cross-origin
+## Use without cross-origin headers
 While not recommended, it is possible to use FRB WASM without using cross-origin isolation.
 Such as in rare cases where HTTPS hosting is not an option.
 In these cases most browsers (for security reasons) will ignore cross-origin headers unless a valid HTTPS connection is made.

--- a/website/docs/manual/miscellaneous/02-web-cross-origin.md
+++ b/website/docs/manual/miscellaneous/02-web-cross-origin.md
@@ -3,10 +3,13 @@
 ## Background
 
 When using Rust (WASM) and Flutter on the web platform,
-the web server needs to respond with the following headers:
+the web server needs to respond with the following headers to enable shared buffers:
 
 - `Cross-Origin-Opener-Policy`: `same-origin`
 - `Cross-Origin-Embedder-Policy`: `credentialless` OR `require-corp` (Safari seems to need `require-corp`)
+
+Cross-origin isolated documents have less restrictions on advanced features, such as asynchronous WASM which utilize shared buffers.
+You can read more about crossOriginIsolation [here](https://developer.mozilla.org/en-US/docs/Web/API/Window/crossOriginIsolated).
 
 ## When `flutter run`
 
@@ -57,6 +60,19 @@ Thanks to [this pull request](https://github.com/flutter/flutter/pull/136297), w
 flutter drive --web-header=Cross-Origin-Opener-Policy=same-origin --web-header=Cross-Origin-Embedder-Policy=require-corp
 ```
 
-## When deploy
+## When `flutter run` without cross-origin
+While not recommended, it is possible to use FRB WASM without using cross-origin isolation.
+Such as in rare cases where HTTPS hosting is not an option.
+In these cases most browsers (for security reasons) will ignore cross-origin headers unless a valid HTTPS connection is made.
+To avoid this we can restrict ourselves to only using the main thread, and making sure we don't try and spawn any more.
+This will stop the need for shared buffers, and likewise will not need the cross-origin headers sent by the web server.
 
+In rust
+- `pub async fn`: On main thread (when in web)
+- `#[frb(sync)] pub fn`: On main thread
+- `pub fn`: NOT on the main thread by default
+
+Alternatively use option `default_dart_async: false` to avoid annotating each function with `#[frb(sync)]`
+
+## When deploy
 Please refer to the web server you are using to see how to add these HTTP headers.

--- a/website/docs/manual/miscellaneous/02-web-cross-origin.md
+++ b/website/docs/manual/miscellaneous/02-web-cross-origin.md
@@ -60,19 +60,22 @@ Thanks to [this pull request](https://github.com/flutter/flutter/pull/136297), w
 flutter drive --web-header=Cross-Origin-Opener-Policy=same-origin --web-header=Cross-Origin-Embedder-Policy=require-corp
 ```
 
-## Use without cross-origin headers
-While not recommended, it is possible to use FRB WASM without using cross-origin isolation.
-Such as in rare cases where HTTPS hosting is not an option.
-In these cases most browsers (for security reasons) will ignore cross-origin headers unless a valid HTTPS connection is made.
-To avoid this we can restrict ourselves to only using the main thread, and making sure we don't try and spawn any more.
-This will stop the need for shared buffers, and likewise will not need the cross-origin headers sent by the web server.
+## Run without cross-origin headers
 
-In rust
-- `pub async fn`: On main thread (when in web)
-- `#[frb(sync)] pub fn`: On main thread
-- `pub fn`: NOT on the main thread by default
+While not recommended, it is possible to use flutter_rust_bridge Web (WASM) without using cross-origin isolation
+by following the steps below.
 
-Alternatively use option `default_dart_async: false` to avoid annotating each function with `#[frb(sync)]`
+When will this be needed:
+For example, when we are in rare cases where HTTPS hosting is not an option, 
+most browsers will ignore cross-origin headers for security reasons.
+
+The main idea is to restrict ourselves to only use the main thread and never use other threads,
+which will stop the need for shared buffers, thus removing the need of cross-origin headers.
+
+To achieve this, simply configure `default_dart_async: false` (which makes every `pub fn` automatically `#[frb(sync)] pub fn`), and then use `pub fn / pub async fn` as usual.
+
+Note that Rust's async system, such as `pub async fn` and [async spawns](../../guides/cross-platform/async), still works well here, because it runs on the main thread by default on the Web platform (though it is multi-threaded on native platforms).
 
 ## When deploy
+
 Please refer to the web server you are using to see how to add these HTTP headers.

